src/coreclr/jit/codegenxarch.cpp

```diff
@@ -7442,8 +7454,10 @@ void CodeGen::genFloatToIntCast(GenTree* treeNode)
     noway_assert((dstSize == EA_ATTR(genTypeSize(TYP_INT))) || (dstSize == EA_ATTR(genTypeSize(TYP_LONG))));
 
     // We shouldn't be seeing uint64 here as it should have been converted
-    // into a helper call by either front-end or lowering phase.
-    noway_assert(!varTypeIsUnsigned(dstType) || (dstSize != EA_ATTR(genTypeSize(TYP_LONG))));
+    // into a helper call by either front-end or lowering phase, unless we have AVX512F
+    // accelerated conversions.
+    noway_assert(!varTypeIsUnsigned(dstType) || (dstSize != EA_ATTR(genTypeSize(TYP_LONG))) ||
+                 compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F));
```

Comment:
This one should be `compIsaSupportedDebugOnly` instead, since its in an assert.

Reply:
yes. Since this is an assert we are supposed to use ```compIsaSupportedDebugOnly```. Changing it accordingly.

Reply:
Aah.. I don't think we can change this because these are noway_asserts and not asserts. ```compIsaSupportedOnly``` is available only in debug mode and not release mode which will lead to failure of release builds.

Reply:
The `compOpportunisticallyDependsOn` checks aren't strictly "cheap" and have "side effects" in the context of some compilers (they track the query and mark the method as using the ISA).

We should make it a regular assert instead so that it only happens in debug builds.

Reply:
Changed noway_assert to assert to make sure comOpportunisticallyDepends On checks are not run unnecessarily.